x/oracle/legacy/v2/store.go

```diff
@@ -0,0 +1,56 @@
+package v2
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/shentufoundation/shentu/v2/x/oracle/types"
+)
+
+func MigrateTaskStore(ctx sdk.Context, storeKey sdk.StoreKey, cdc codec.BinaryCodec) error {
+	store := ctx.KVStore(storeKey)
+	expireHeightTaskIDs := make(map[int64][]types.TaskID)
+	iterator := sdk.KVStorePrefixIterator(store, types.TaskStoreKeyPrefix)
+
+	defer iterator.Close()
+	for ; iterator.Valid(); iterator.Next() {
+		var oldTask Task
+		cdc.MustUnmarshalLengthPrefixed(iterator.Value(), &oldTask)
+
+		newTask := types.NewTask(
+			oldTask.Contract,
+			oldTask.Function,
+			oldTask.BeginBlock,
+			oldTask.Bounty,
+			oldTask.Description,
+			oldTask.Expiration,
+			sdk.AccAddress(oldTask.Creator),
+			oldTask.ClosingBlock,
+			oldTask.WaitingBlocks,
+		)
+		newTask.Status = types.TaskStatus(oldTask.Status)
+
+		// delete old task
+		store.Delete(iterator.Key())
+		// set task
+		bz, err := cdc.MarshalInterface(&newTask)
+		if err != nil {
+			panic(err)
+		}
+		store.Set(types.TaskStoreKey(newTask.GetID()), bz)
+		// get all ExpireHeight and taskIDs
+		if newTask.Status == types.TaskStatusPending {
+			expireHeight := newTask.ExpireHeight
+			newTaskID := types.TaskID{Tid: newTask.GetID()}
+			expireHeightTaskIDs[expireHeight] = append(expireHeightTaskIDs[expireHeight], newTaskID)
+		}
+	}
+
+	//  Migrate ExpireHeight and taskIDs
+	for height, taskIDs := range expireHeightTaskIDs {
```

Comment:
Can we make sure there is no legacy ClosingTaskIDs being left over?

Reply:
I used another method to ensure zero loss. By traversing the height, I obtained the `ClosingTaskIDs` in the database and then parsed them for conversion.